feat: Panelplot config redesign (dashboard)

vwrj opened the pull request:
#### Loom walkthrough: 
https://www.loom.com/share/5d39b8212bc8425784c133003fe6aa17?sid=7c3c4097-7afe-447f-a6ad-283ac5aec084

#### Screenshots:

<img width="1102" alt="Screen Shot 2023-07-19 at 12 29 38 AM" src="https://github.com/wandb/weave/assets/8609620/e063470c-9f12-43aa-b37c-85afaf75a453">

---

<img width="1088" alt="Screen Shot 2023-07-19 at 12 30 24 AM" src="https://github.com/wandb/weave/assets/8609620/55e1002f-174b-46af-91f8-8c8d90e2ef67">

---

<img width="1093" alt="Screen Shot 2023-07-19 at 12 29 59 AM" src="https://github.com/wandb/weave/assets/8609620/abc2210c-b834-4591-9762-29d1a1d5996e">

dannygoldstein commented on weave-js/src/components/Panel2/PanelPlot/PanelPlot.tsx:
```diff
@@ -595,6 +656,119 @@ const PanelPlotConfigInner: React.FC<PanelPlotProps> = props => {
+  //         extractWeaveExpressionDims(seriesDim).map(dim => {
+  //           dimensions.push({name: `${dimName}-${i + 1}`, dim});
+  //         });
+  //       }
+  //     }
+  //   }
+  //   return dimensions;
+  // }, [config, weave]);
+
+  // const configGroupedDims = useMemo(() => {
+  //   const groupedDims: string[] = [];
+  //   config.series.map((s, i) => {
+  //     s.table.groupBy.map(col => {
+  //       for (let key in s.dims) {
+  //         const value = s.dims[key as keyof DIM_NAME_MAP] as string;
+  //         if (value === col) {
+  //           groupedDims.push(`${key}-${i + 1}`);
+  //         }
+  //       }
+  //     });
+  //   });
+  //   return groupedDims;
+  // }, [config]);
+
+  // const groupByDom = useMemo(() => {
+  //   return (
+  //     <ConfigPanel.ConfigOption label="Grouping">
+  //       <ConfigPanel.ModifiedDropdownConfigField
+  //         multiple
+  //         options={groupByDims.map(d => {
+  //           return {key: d.name, text: d.name, value: d.name};
+  //         })}
+  //         value={configGroupedDims}
+  //         onChange={(event, {value}) => {
+  //           console.log('hey dropdown value set by user', value);
+  //         }}
+  //       />
+  //     </ConfigPanel.ConfigOption>
+  //   );
+  // }, [groupByDims, config]);
```
we'll want this in the diff for the next pr

vwrj replied:
I'm going to work on this right after this PR so I thought it'd be easier just to have the commented out portion there for easy access

dannygoldstein replied:
can you just make a new branch with this in there? otherwise when I look at your next PR there wont be a diff. and i wont know what went in related to grouping without manually looking in the code

dannygoldstein replied:
nvm. i guess there will be a diff with the comment // removed

vwrj replied:
moved into new branch

dannygoldstein commented on weave-js/src/components/Panel2/PanelPlot/PanelPlot.tsx:
```diff
@@ -477,30 +480,88 @@ const PanelPlotConfigInner: React.FC<PanelPlotProps> = props => {
+                    const newConfig = PlotState.addSeriesFromSeries(
+                      config,
+                      s,
+                      'x',
```
why x? i think y is likely the best choice to use here

dannygoldstein commented on weave-js/src/components/Panel2/PanelPlot/PanelPlot.tsx:
```diff
@@ -477,30 +480,88 @@ const PanelPlotConfigInner: React.FC<PanelPlotProps> = props => {
+                  },
+                },
+                {
+                  key: 'Add series from this series',
```
In @Cecile0112358's design this was a button displayed more prominently, not something in an overflow menu, did that change? 

https://www.figma.com/file/Vc1PpAp2ymWTJPbsj444D0/Weave-1.1?type=design&node-id=5123-111782&mode=design&t=dwBpZGayGtau1pqA-0

vwrj replied:
do we have a method that creates a new empty Series from scratch? I'm only seeing `PlotState.addSeriesFromSeries `which requires a pre-existing series.

dannygoldstein replied:
Decided on slack to add a button which adds series from latest series

vwrj replied:
added

dannygoldstein commented on weave-js/src/components/Panel2/PanelPlot/PanelPlot.tsx:
```diff
@@ -477,30 +480,88 @@ const PanelPlotConfigInner: React.FC<PanelPlotProps> = props => {
+                      'x',
+                      weave
+                    );
+                    updateConfig(newConfig);
```
before  we call updateConfig here, we should set the new series name to something different from the original. maybe a default like "Series [N]" or something

vwrj replied:
It already increments and creates "Series {N+1}" if there are currently N series.

dannygoldstein replied:
Thats the label for the config component, but not the actual series name, which lives in config.series[n].name

vwrj replied:
I'm printing out the config object and I'm not seeing the name attribute on any series that I create, even on the first one. Where does it exist?

dannygoldstein replied:
Its on `config.series[n].name`

dannygoldstein replied:
https://github.com/wandb/weave/blob/master/weave-js/src/components/Panel2/PanelPlot/versions/v6.ts#L18

vwrj replied:
added

dannygoldstein reviewed (commented):
We need to keep the styling in the app the same, it seems to have changed a bit with this PR: 

This is what it looks like with this PR: 

<img width="463" alt="Screen Shot 2023-07-19 at 9 35 56 AM" src="https://github.com/wandb/weave/assets/2769632/e680292d-34ec-47ff-bf93-84dfbed39948">

And this is how it should look:

<img width="444" alt="Screen Shot 2023-07-19 at 9 43 10 AM" src="https://github.com/wandb/weave/assets/2769632/9974b773-65aa-4cdd-989a-7397dfcf04c3">

Issues in that screenshot diff that need to be fixed: 

1. Preserve series / dimension numbering in App
2. No vertical line next to series in app


Source: https://wandb.ai/stacey/mdtime/reports/test-weather-forecasting--VmlldzoyNzM2Njg5, panel1

vwrj commented on weave-js/src/components/Panel2/ConfigPanel/index.tsx:
```diff
@@ -77,9 +86,25 @@ export const ConfigSection: FC<ConfigSectionProps> = ({label, children}) => {
-          <ConfigSectionHeaderButton expanded={expanded}>
-            <IconCaret />
-          </ConfigSectionHeaderButton>
+          <div style={{display: 'flex'}}>
+            {menuItems != null && (
+              <PopupMenu
+                position="bottom left"
+                trigger={
+                  <ConfigDimMenuButton
+                    onClick={e => {
+                      e.stopPropagation();
+                    }}>
+                    <IconOverflowHorizontal />
+                  </ConfigDimMenuButton>
+                }
+                items={menuItems}
+              />
+            )}
+            <ConfigSectionHeaderButton expanded={expanded}>
+              <IconCaret />
+            </ConfigSectionHeaderButton>
+          </div>
```
had a huddle about this, everything seems to be good. display: flex with one element works fine.

dannygoldstein reviewed (commented):
Looking better! Couple more things to fix: 

App comments:
1. This PR alters the way that the add new series from series button works in the app. Before, when you add a new series from a series you get a blank expression editor on the dimension that you added. Now it copies over the state from that dimension. We should ensure that, in the app, the old behavior is preserved. Its fine to copy over in dashboards. 

Dashboard comments:
2. In dashboards, we should set `series.name` to some default when we add a new series. Right now it gives a null name for each series which is rendered as the same empty text editor for all series
3. We should hide the constrain dimension button when there is only one series. Right now it shows up even when there is only one series, and is a no-op there. 
4. We should hide the advanced properties menu when there are no advanced properties to show (like if you have an empty plot config for example)
5. The constraint button appears in derivative dimensions like `Style`, 
<img width="341" alt="Screen Shot 2023-07-20 at 2 52 22 PM" src="https://github.com/wandb/weave/assets/2769632/a78efa41-58bd-486c-94e7-bf747154fd39">. We should get rid of it there. Only top level dimensions should have the option to be constrained
6. Updating the color legend label does not appear to have any effect

vwrj commented:
> 

1. Chatted in Slack, this is a problem in prod too so nothing particular to this PR. 
2. fixed
3. fixed
4. fixed
5. fixed
6. Chatted in Slack, this is the same behavior as in prod, nothing particular to this PR
